src/org/opendatakit/briefcase/ui/automation/AutomationPanelForm.java

```diff
@@ -0,0 +1,138 @@
+package org.opendatakit.briefcase.ui.automation;
+
+import static org.opendatakit.briefcase.ui.reused.FileChooser.directory;
+
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+import org.opendatakit.briefcase.util.StringUtils;
+
+@SuppressWarnings("checkstyle:MethodName")
+public class AutomationPanelForm {
+  public JPanel container;
+  private JLabel scriptDirLabel;
+  private JTextField scriptDirField;
+  private JButton scriptDirChooseButton;
+  private JButton generateScriptButton;
+
+  public AutomationPanelForm() {
+    $$$setupUI$$$();
+    scriptDirChooseButton.addActionListener(__ ->
+        directory(container, fileFrom(scriptDirField))
+            .choose()
+            .ifPresent(file -> setScriptDir(Paths.get(file.toURI()))));
+    generateScriptButton.addActionListener(__ -> {
```

Comment:
The code on this block belongs to the AutomationPanel. The view should not know anything about how the scripts are generated. It should just let the panel subscribe to the "generate script" business event generate an `AutomationConfiguration` from the values the user has entered using the UI

Reply:
Makes sense. But I thought the AutomationPanel will be come with a configuration object which will make the onGenerate method take a Runnable instead of a Consumer<AutomationConfiguration> that way the onGenerate event handler will rely on the configuration that came with it instead of the one been sent by the AutomationPanelForm and the signature will be 

```java
view.onGenerate(() -> {

});
```
Instead of

```java
view.onGenerate(config -> {

});
```

Similar to how it's done here https://github.com/opendatakit/briefcase/blob/master/src/org/opendatakit/briefcase/ui/export/ExportPanel.java#L87-L97
```java
form.onExport(() -> {
      form.setExporting();
      List<String> errors = getErrors();
      if (errors.isEmpty()) {
        new Thread(this::export).start();
      } else {
        analytics.event("Export", "Export", "Configuration errors", null);
        showErrorDialog(getForm().getContainer(), errors.stream().collect(joining("\n")), "Export errors");
        form.unsetExporting(appPreferences.getRememberPasswords().orElse(false));
      }
    });
```

Reply:
I know this can be confusing. The export tab is the first we refactored and I tried some stuff that I've been improving on newer tab refactors. 

My current rationale about this kind of UI components is that they produce the conf object and let the controllers subscribe with a consumer that will use the conf to do stuff.

Newer tabs like settings, pull and push work that way.

That's why suggested to have an `onGenerate(Consumer<AutomationConfiguration> callback)`.

The export tab doesn't work that way and it's way more complex than the other tabs because of this.

Does that make sense?